syncer/syncer.go

```diff
@@ -0,0 +1,285 @@
+package syncer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/0xPolygon/polygon-edge/helper/progress"
+	"github.com/0xPolygon/polygon-edge/network/event"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+const (
+	syncerLoggerName = "syncer"
+	SyncerProto      = "/syncer/0.2"
+)
+
+var (
+	errTimeout = errors.New("timeout awaiting block from peer")
+)
+
+// XXX: Don't use this syncer for the consensus that may cause fork.
+// This syncer doesn't assume fork. Consensus may be broken.
+// TODO: Add extensibility for fork before merge
```

Comment:
I take it we won't be having consensuses that are not instant finality in a long time. Can this TODO be removed ? (and everything related to forks)

Reply:
Actually this comment is not for us, but someone who wants to extend Edge and use it

Reply:
@dbrajovic

Reply:
I think module must not depend on the implementation of the module. Unfortunately syncer v2 completely doesn't work with the consensus that doesn't guarantee immediate finality. This is breaking modularity completely. Initially syncer v2 had such an extensibility but stop it because it made the code complex.

Reply:
For that reason, I've left such comments

Reply:
Comments will be for people with various aspects, future engineer at Polygon Edge, user, and third-party who wants extend Polygon Edge.

Reply:
Agree with @Kourin1996, definition of TODO tag is

_In general, a TODO is written to tell future maintainers about something important: something that should be added, or should be changed, or should not be forgotten._

@dbrajovic  But also we need to rephrase this because we will merge the syncer module without adding support for the fork.
(maybe remove **before merge** from the sentence)